workers/data_refinery_workers/processors/agilent_twocolor.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import absolute_import, unicode_literals
+
+import os
+import string
+import warnings
+from django.utils import timezone
+from typing import Dict
+
+import rpy2.robjects as ro
+from rpy2.rinterface import RRuntimeError
+
+from data_refinery_common.logging import get_and_configure_logger
+from data_refinery_common.models import OriginalFile, ComputationalResult, ComputedFile, SampleResultAssociation
+from data_refinery_workers._version import __version__
+from data_refinery_workers.processors import utils
+from data_refinery_common.utils import get_env_variable
+S3_BUCKET_NAME = get_env_variable("S3_BUCKET_NAME", "data-refinery")
+
+logger = get_and_configure_logger(__name__)
+
+
+def _prepare_files(job_context: Dict) -> Dict:
+    """Populate our job_context with appropriate inputs and outputs
+
+    Also adds the keys "input_file_path" and "output_file_path" to
+    job_context so everything is prepared for processing.
+    """
+    original_file = job_context["original_files"][0]
+    job_context["input_file_path"] = original_file.absolute_file_path
+    # Turns /home/user/data_store/E-GEOD-8607/raw/foo.txt into /home/user/data_store/E-GEOD-8607/processed/foo.cel
+    pre_part = original_file.absolute_file_path.split('/')[:-2]
+    end_part = original_file.absolute_file_path.split('/')[-1]
+
+    os.makedirs('/'.join(pre_part) + '/processed/', exist_ok=True)
+    job_context["output_file_path"] = '/'.join(pre_part) + '/processed/' + end_part
+    job_context["output_file_path"] = job_context["output_file_path"].replace('.txt', '.PCL')
+
+    return job_context
+
+def _run_scan_twocolor(job_context: Dict) -> Dict:
+    """Processes an input TXT file to an output PCL file.
+
+    Does so using the SCAN.UPC package's SCAN_TwoColor method using R.
+    Expects job_context to contain the keys 'input_file' and 'output_file'.
+    """
+    input_file = job_context["input_file_path"]
+
+    try:
+        # It's necessary to load the foreach library before calling SCAN_TwoColor
+        # because it doesn't load the library before calling functions
+        # from it.
+        ro.r("suppressMessages(library('foreach'))")
+
+        # Prevents:
+        # RRuntimeWarning: There were 50 or more warnings (use warnings()
+        # to see the first 50)
+        ro.r("options(warn=1)")
+
+        # All R messages are turned into Python 'warnings' by rpy2. By
+        # filtering all of them we silence a lot of useless output
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            scan_upc = ro.r['::']('SCAN.UPC', 'SCAN_TwoColor')
+            job_context['time_start'] = timezone.now()
+
+            # XXX: Current bug lives here. See below.
+            scan_upc(input_file,
```

Comment:
:+1: these are the only two arguments that should be going to `SCAN.UPC::SCAN_TwoColor`